### ISSUE TITLE
Allow Calamares to set GRUB password

### ISSUE THREAD

**Issue by UriHerrera** [calamares/calamares#2137]
**Is your feature request related to a problem? Please describe.**
- In Calamares 3.2.61, setting a password for GRUB is impossible.

**Describe the solution you'd like**
- I think that either reusing the password for the user that is created during the installation or the root password if enabled as the password for GRUB would be a simple solution.

**Describe alternatives you've considered**
- For now, to set a password for GRUB I've used shellprocess.

**Comment by demmm** [comment]
>For now, to set a password for GRUB, I've used the shellprocess module.

If you already have this feature working, why not create a PR and add your solution to grubcfg?

**Comment by UriHerrera** [comment]
> > For now, to set a password for GRUB, I've used the shellprocess module.
> 
> If you already have this feature working, why not create a PR and add your solution to grubcfg?

Because what I'd like to be able to do is something like this.

```
setGRUBPassword: true
useUserPassword: true
```

Instead of using shellprocess and doing this.

```
script:
    - "bash @@ROOT@@/bin/script-to-set-grub-password"
```

**Comment by adrelanos** [comment]
Is this still needed?

calamares (by Debian bookworm) can now set grub to unlock the fully encrypted (including /boot) hard drive. grub is already asking for a password at boot time.

So it seems to me that this is already implemented?

**Comment by dalto8** [comment]
> calamares (by Debian bookworm) can now set grub to unlock the fully encrypted (including /boot) hard drive.

Isn't this about setting a grub password?  I think what you are referring to is grub requesting a password to unlock an encrypted partition.

**Comment by UriHerrera** [comment]
> Is this still needed?
> 
> calamares (by Debian bookworm) can now set grub to unlock the fully encrypted (including /boot) hard drive. grub is already asking for a password at boot time.
> 
> So it seems to me that this is already implemented?

Yes. The issue concerns Calamares allowing the user to set a GRUB password; the password you're referencing is for partitions using cryptsetup.

**Comment by adrelanos** [comment]
If you can encrypt all including /boot, no more need for unencrypted boot + grub password?

**Comment by UriHerrera** [comment]
> If you can encrypt all including /boot, no more need for unencrypted boot + grub password?

@adrelanos, I'll add more context to the use case of why I opened this issue.

Currently, I'm using Calamares 3.2.62, which is newer than the version packaged in Debian 12, which is 3.2.61-1+b2. I'm not using Legacy BIOS and MBR but Legacy BIOs, EFI, and GPT. Nor am I using a single partition layout, where `/boot` would be a directory within the root.

I do not have the option to encrypt the `boot` partition in the GUI whether I want to encrypt an existing partition or when creating a new partition. Depending on the selection of the filesystem, the checkbox to encrypt the partition appears. It does not appear with `fat32` or `fat16`; see images below.

![Captura de pantalla de 2023-07-31 20-05-01](https://github.com/calamares/calamares/assets/3053525/1f108a68-10a3-49ac-89eb-2fdeddf0a396)
> Existing partition.

![Captura de pantalla de 2023-07-31 20-05-28](https://github.com/calamares/calamares/assets/3053525/ac828294-1ca3-4712-bdc7-e86cdc9e473c)
> New partition EFI.

![Captura de pantalla de 2023-07-31 20-29-26](https://github.com/calamares/calamares/assets/3053525/becb9558-b567-47da-a0d4-24fbdca3fbac)
> New partition Legacy BIOS.

Also, I do not see a key in the module configuration file (`partition.conf`) to encrypt that partition. I only see `enableLuksAutomatedPartitioning`, which I've set to `true`. I do not know if I don't see the checkbox to encrypt the `/boot` partition because I'm using GRUB, not systemd-boot.

So, from what I can see since I can't encrypt the `/boot` partition, there is no way to set a password for GRUB when using Calamares. Thus, as I mentioned here, https://github.com/calamares/calamares/issues/2137#issuecomment-1546887936, I'd like to be able to set a key instead of using a script.

**Comment by adrelanos** [comment]
It is important to make sure that we do converse about the same topic.

----

_Figure:_ **GRUB Password Prompt for Disk Decryption (EFI)**

![EFIGrubpass](https://github.com/user-attachments/assets/cb054b59-8770-47e9-96cc-a4ea49e198eb)

* This prompt is related to unlocking an **encrypted `/boot` partition** during system startup.
* (Shown here within VirtualBox; the appearance will be similar on physical hardware.)

---

_Figure:_ **GRUB Authentication Password Prompt**

![grub_bootloader_password](https://github.com/user-attachments/assets/6f849851-7d78-434e-8aea-5e0eb3a0c2bd)

* This prompt enforces user authentication for accessing GRUB menu entries and bootloader configurations.
* Unrelated to encrypted `/boot`.
* More details: [Authentication and authorisation in GRUB](https://www.gnu.org/software/grub/manual/grub/html_node/Authentication-and-authorisation.html)

----

Is this feature request about:

* A) GRUB Password Prompt for Disk Decryption, or
* B) GRUB Authentication Password Prompt?
